esmvalcore/cmor/_fixes/fix.py

```diff
@@ -202,11 +226,15 @@ def get_fixes(project, dataset, mip, short_name, extra_facets=None):
                 pass
 
         for fixes_module in fixes_modules:
-            classes = inspect.getmembers(fixes_module, inspect.isclass)
-            classes = dict((name.lower(), value) for name, value in classes)
+            classes = dict(
+                (name.lower(), value) for (name, value) in
+                inspect.getmembers(fixes_module, inspect.isclass)
+            )
             for fix_name in (short_name, mip.lower(), 'allvars'):
                 try:
-                    fixes.append(classes[fix_name](vardef, extra_facets))
+                    fixes.append(classes[fix_name](
+                        vardef, extra_facets=extra_facets, session=session
+                    ))
                 except KeyError:
```

Comment:
note that `session` may fail with a key error here too but the exception is supposed to catch the `fix_name` key error only and not other stuff, maybe call session once before if not None to catch anything bad from it?

Reply:
`session` is already initialized here, but I agree that it's probably better to convert the `try/except` to a `if fix_name in classes`. Got some weird error messages lately because of a similar problem.